library/common/event/provisional_dispatcher.cc

```diff
@@ -25,8 +25,12 @@ envoy_status_t ProvisionalDispatcher::post(Event::PostCb callback) {
   Thread::LockGuard lock(state_lock_);
 
   if (drained_) {
-    event_dispatcher_->post(callback);
-    return ENVOY_SUCCESS;
+    if (event_dispatcher_) {
```

Comment:
If this is for shutdown, when does this get set to null? I wouldn't expect the destructor to null this out 

If this is post-destruction I think ideally you want to avoid calling `post` at all since we're in undefined behavior territory

Reply:
ok that makes sense, but how could I detect that we're post-destruction at the call site here? https://github.com/envoyproxy/envoy-mobile/blob/c93132cbe4b05063c6c6f307666eb49d029daea3/library/common/engine_handle.cc#L8

Reply:
The C++ way is to never have handles to anything that is deallocated, since there's really no way to inspect something post-destruction to see if destruction has happened (this is UB if done naively). What I would probably want to understand is how we can lock the shared ptr but end up with a handle that is not guaranteed to be valid for the duration of this shared ptr. If we were calling `terminateEngine` I can see how this happens (we call terminate even though there might be external references to the engine), but I think in the cases we're seeing the issue we're not explicitly terminating? So not sure how this plays out in that case

Reply:
ok I'll think about this some more and re-open a new PR if I come up with a good way to handle this.